bib/internal/progress/progress.go

```diff
@@ -0,0 +1,296 @@
+// lower level osbuild/images message. It will never clear the screen
+// so "glitches/weird" messages from the lower-layers can be inspected
+// easier.
+func NewDebugProgressBar() (ProgressBar, error) {
+	np := &debugProgressBar{w: osStderr}
+	return np, nil
+}
+
+func (np *debugProgressBar) SetPulseMsg(msg string, args ...interface{}) {
+	fmt.Fprintf(np.w, "pulse: ")
+	fmt.Fprintf(np.w, msg, args...)
+	fmt.Fprintf(np.w, "\n")
+}
+
+func (np *debugProgressBar) SetMessage(msg string, args ...interface{}) {
+	fmt.Fprintf(np.w, "msg: ")
+	fmt.Fprintf(np.w, msg, args...)
+	fmt.Fprintf(np.w, "\n")
+}
+
+func (np *debugProgressBar) Start() (err error) {
+	fmt.Fprintf(np.w, "Start progressbar\n")
+	return nil
+}
+
+func (np *debugProgressBar) Stop() (err error) {
+	fmt.Fprintf(np.w, "Stop progressbar\n")
+	return nil
+}
+
+func (np *debugProgressBar) SetProgress(subLevel int, msg string, done int, total int) error {
+	fmt.Fprintf(np.w, "%s[%v / %v] %s", strings.Repeat("  ", subLevel), done, total, msg)
+	fmt.Fprintf(np.w, "\n")
+	return nil
+}
+
+// XXX: merge variant back into images/pkg/osbuild/osbuild-exec.go
+func RunOSBuild(pb ProgressBar, manifest []byte, store, outputDirectory string, exports, extraEnv []string) error {
+	switch pb.(type) {
+	case *termProgressBar, *debugProgressBar:
+		return runOSBuildNew(pb, manifest, store, outputDirectory, exports, extraEnv)
+	default:
+		return runOSBuildOld(pb, manifest, store, outputDirectory, exports, extraEnv)
+	}
+}
+
+func runOSBuildOld(pb ProgressBar, manifest []byte, store, outputDirectory string, exports, extraEnv []string) error {
+	_, err := osbuild.RunOSBuild(manifest, store, outputDirectory, exports, nil, extraEnv, false, os.Stderr)
+	return err
+}
+
+func runOSBuildNew(pb ProgressBar, manifest []byte, store, outputDirectory string, exports, extraEnv []string) error {
+	rp, wp, err := os.Pipe()
+	if err != nil {
+		return fmt.Errorf("cannot create pipe for osbuild: %w", err)
+	}
+	defer rp.Close()
+	defer wp.Close()
+
+	cmd := exec.Command(
+		"osbuild",
+		"--store", store,
+		"--output-directory", outputDirectory,
+		"--monitor=JSONSeqMonitor",
+		"--monitor-fd=3",
+		"-",
+	)
+	for _, export := range exports {
+		cmd.Args = append(cmd.Args, "--export", export)
+	}
+
+	cmd.Env = append(os.Environ(), extraEnv...)
+	cmd.Stdin = bytes.NewBuffer(manifest)
+	cmd.Stderr = os.Stderr
+	// we could use "--json" here and would get the build-result
+	// exported here
+	cmd.Stdout = nil
+	cmd.ExtraFiles = []*os.File{wp}
+
+	osbuildStatus := osbuild.NewStatusScanner(rp)
+	if err := cmd.Start(); err != nil {
+		return fmt.Errorf("error starting osbuild: %v", err)
+	}
+	wp.Close()
+
+	var tracesMsgs []string
+	for {
+		st, err := osbuildStatus.Status()
+		if err != nil {
+			return fmt.Errorf("error reading osbuild status: %w", err)
+		}
+		if st == nil {
+			break
+		}
+		i := 0
+		for p := st.Progress; p != nil; p = p.SubProgress {
+			// XXX: osbuild gives us bad progress messages
```

Comment:
What does this mean?

Reply:
Sorry, I think this is not actually the case anymore, I fixed this (removed it)